thinc/layers/remap_ids.py

```diff
@@ -1,41 +1,99 @@
-from typing import Tuple, Callable, Sequence, Dict, Any
+from typing import Tuple, Callable, Sequence, cast
+from typing import Dict, Union, Optional
 
 from ..model import Model
 from ..config import registry
-from ..types import Ints2d, DTypes
+from ..types import Ints1d, Ints2d, ArrayXd
+from ..util import is_xp_array, to_numpy
 
 
-InT = Sequence[Any]
+InT = Union[Sequence[str], Sequence[int], Ints1d, Ints2d]
 OutT = Ints2d
+IntVecOrMat = Union[Ints1d, Ints2d]
```

Comment:
To follow our usual naming conventions, something more like:

```suggestion
Ints1dOr2d = Union[Ints1d, Ints2d]
```